app/Commands/NewCommand.php

```diff
@@ -0,0 +1,474 @@
+
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        $this->line($process->getOutput());
+        return true;
+    }
+
+    protected function doGit()
+    {
+        if ($this->option('message')) {
+            $this->commitmessage = $this->option('message');
+        }
+
+        if (! $this->tools['git']) {
+            $this->info("Unable to find 'git' on the system so I cannot initialize a git repo in '{$this->projectpath}'");
+
+            return false;
+        }
+
+        $process = new Process('dummy command');
+        $process->setWorkingDirectory($this->projectpath);
+
+        $commands = [
+            'git init',
+            'git add .',
+            'git commit -m "'.str_replace('"', '\"', $this->commitmessage).'"',
+        ];
+
+        foreach ($commands as $command) {
+            $process->setCommandLine($command);
+            $process->run();
+            $this->line($process->getOutput());
+        }
+
+        return true;
+    }
+
+    protected function openTextEditor()
+    {
+        if ($this->option('editor')) {
+            $editor = $this->option('editor');
+        } else {
+            $finder = new ExecutableFinder();
+            foreach ($this->editors_gui as $_editor) {
+                if ($finder->find($_editor)) {
+                    $editor = $_editor;
+                    break;
+                }
+            }
+            if (empty($editor)) {
+                foreach ($this->editors_terminal as $_editor) {
+                    if ($finder->find($_editor)) {
+                        $editor = $_editor;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (empty($editor)) {
+            $this->warn('Unable to find a text editor to open, skipping this step.');
+
+            return false;
+        }
+
+        $this->info("Found editor $editor so an opening it now");
+
+        $process = new Process("$editor .");
+        $process->setWorkingDirectory($this->projectpath);
+
+        $process->setTty(true);
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        return true;
+    }
+
+    protected function createDatabase($type = 'sqlite')
+    {
+        if ($type === 'sqlite') {
+            $basedir = $this->projectpath.DIRECTORY_SEPARATOR.'database';
+
+            // @todo Could not find nice way to add (touch) an empty file here with Filesystem so am doing it manually
+            $process = new Process('touch database.sqlite');
+            $process->setWorkingDirectory($basedir);
+            $process->run();
+
+            $this->info($process->getOutput());
+
+            $this->replaceEnvVariable('DB_CONNECTION', 'sqlite');
+            // @todo Seems this must be a full path, true? Also, should we use a config/ file instead of .env?
```

Comment:
hm. Good question. I'm not sure.

Reply:
Keeping as a full path, will investigate later as we consider and discuss using `config/` for things instead of only `.env`.